In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
#!pip install pyarrow

In [3]:
# for year in range(3, 24):
#     # Formating the year to have two digits
#     year_str = f"{year:02d}"
    
#     # Parquet path
#     parquet_file = f'malaria_notification_data/NOTIPO{year_str}.parquet'
    
#     # CSV path
#     csv_file = f'malaria_notification_data/NOTIPO{year_str}.csv'
    
#     parquet_df = pd.read_parquet(parquet_file)
#     parquet_df.to_csv(csv_file, index=False)

In [2]:
malaria_2003_df = pd.read_csv('malaria_notification_data_csv/NOTIPO03.csv')
malaria_2004_df = pd.read_csv('malaria_notification_data_csv/NOTIPO04.csv')
malaria_2005_df = pd.read_csv('malaria_notification_data_csv/NOTIPO05.csv')
malaria_2006_df = pd.read_csv('malaria_notification_data_csv/NOTIPO06.csv')
malaria_2007_df = pd.read_csv('malaria_notification_data_csv/NOTIPO07.csv')
malaria_2008_df = pd.read_csv('malaria_notification_data_csv/NOTIPO08.csv')
malaria_2009_df = pd.read_csv('malaria_notification_data_csv/NOTIPO09.csv')
malaria_2010_df = pd.read_csv('malaria_notification_data_csv/NOTIPO10.csv')
malaria_2011_df = pd.read_csv('malaria_notification_data_csv/NOTIPO11.csv')
malaria_2012_df = pd.read_csv('malaria_notification_data_csv/NOTIPO12.csv')
malaria_2013_df = pd.read_csv('malaria_notification_data_csv/NOTIPO13.csv')
malaria_2014_df = pd.read_csv('malaria_notification_data_csv/NOTIPO14.csv')
malaria_2015_df = pd.read_csv('malaria_notification_data_csv/NOTIPO15.csv')
malaria_2016_df = pd.read_csv('malaria_notification_data_csv/NOTIPO16.csv')
malaria_2017_df = pd.read_csv('malaria_notification_data_csv/NOTIPO17.csv')
malaria_2018_df = pd.read_csv('malaria_notification_data_csv/NOTIPO18.csv')
malaria_2019_df = pd.read_csv('malaria_notification_data_csv/NOTIPO19.csv')
malaria_2020_df = pd.read_csv('malaria_notification_data_csv/NOTIPO20.csv')
malaria_2021_df = pd.read_csv('malaria_notification_data_csv/NOTIPO21.csv')
malaria_2022_df = pd.read_csv('malaria_notification_data_csv/NOTIPO22.csv')
malaria_2023_df = pd.read_csv('malaria_notification_data_csv/NOTIPO23.csv')

C:\Users\ilana\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (30) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
C:\Users\ilana\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (30,41) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
C:\Users\ilana\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (41) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
C:\Users\ilana\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3071: DtypeWarning: Columns (37,38,41) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,

In [4]:
full_malaria_df = pd.concat([
    malaria_2003_df, malaria_2004_df, malaria_2005_df, malaria_2006_df,
    malaria_2007_df, malaria_2008_df, malaria_2009_df, malaria_2010_df,
    malaria_2011_df, malaria_2012_df, malaria_2013_df, malaria_2014_df,
    malaria_2015_df, malaria_2016_df, malaria_2017_df, malaria_2018_df,
    malaria_2019_df, malaria_2020_df, malaria_2021_df, malaria_2022_df,
    malaria_2023_df])

full_malaria_df = full_malaria_df.reset_index().drop(columns=['index', 'Unnamed: 0'])
full_malaria_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM  \
0              56  27/01/03  23/03/2011       NaN  05/2003         3   
1             470  20/03/03  23/03/2011       NaN  12/2003         2   
2            1309  10/07/03  23/03/2011       NaN  28/2003         1   
3              19  27/01/03  23/03/2011       NaN  05/2003         3   
4             657  04/04/03  23/03/2011       NaN  14/2003         1   
...           ...       ...         ...       ...      ...       ...   
6816637       225  20/07/23  26/09/2023  26/09/23  29/2023         1   
6816638       524  19/06/23  11/07/2023  11/07/23  25/2023         1   
6816639       922  05/12/23  12/12/2023  12/12/23  49/2023         1   
6816640        14  26/09/23  20/10/2023  20/10/23  39/2023         1   
6816641         1  28/08/23  11/03/2024  11/03/24  35/2023         2   

         UF_NOTIF  MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  \
0              21    210290      9505  498068.0  ...                  NaN   
1              21    210232     10033  232514.0  ...                  NaN   
2              21    210232     10032       NaN  ...                  NaN   
3              21    210232     10032  232516.0  ...                  NaN   
4              21    210232    100321       NaN  ...                  NaN   
...           ...       ...       ...       ...  ...                  ...   
6816637        13    130020        49     618.0  ...                  NaN   
6816638        13    130406        46      64.0  ...                  NaN   
6816639        13    130406        24     339.0  ...                  NaN   
6816640        16    160025         1  352826.0  ...                  NaN   
6816641        17    172120   1721201    1789.0  ...                  NaN   

         EXAMINADOR ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  \
0               NaN       1         5          NaN           NaN   
1               NaN       2         6          NaN           NaN   
2               NaN       2       NaN          NaN           NaN   
3               NaN       1         6          NaN           NaN   
4               NaN       2         4          NaN           NaN   
...             ...     ...       ...          ...           ...   
6816637       618.0       2         6          NaN           NaN   
6816638        67.0       2         6          NaN           NaN   
6816639       449.0       2         6          NaN           NaN   
6816640    352826.0       2         6          NaN           NaN   
6816641      1789.0       2         6          NaN           NaN   

         DIMEA_AMAMENT  VIVAX  FALCIPARUM  NIV_ESCO  
0                  NaN    NaN         NaN       0.0  
1                  NaN    NaN         NaN       1.0  
2                  NaN    NaN         NaN       0.0  
3                  NaN    NaN         NaN       NaN  
4                  NaN    NaN         NaN       0.0  
...                ...    ...         ...       ...  
6816637            NaN    2.0         2.0       1.0  
6816638            NaN    2.0         2.0       0.0  
6816639            NaN    2.0         2.0       2.0  
6816640            NaN    2.0         2.0       5.0  
6816641            NaN    2.0         2.0       6.0  

[6816642 rows x 48 columns]

#### Gets vivax cases (RES_EXAM = 4)

In [5]:
vivax_cases_df = full_malaria_df[full_malaria_df.get('RES_EXAM')==4]
vivax_cases_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM  \
0              56  27/01/03  23/03/2011       NaN  05/2003         3   
1             470  20/03/03  23/03/2011       NaN  12/2003         2   
2            1309  10/07/03  23/03/2011       NaN  28/2003         1   
3              19  27/01/03  23/03/2011       NaN  05/2003         3   
4             657  04/04/03  23/03/2011       NaN  14/2003         1   
...           ...       ...         ...       ...      ...       ...   
6816637       225  20/07/23  26/09/2023  26/09/23  29/2023         1   
6816638       524  19/06/23  11/07/2023  11/07/23  25/2023         1   
6816639       922  05/12/23  12/12/2023  12/12/23  49/2023         1   
6816640        14  26/09/23  20/10/2023  20/10/23  39/2023         1   
6816641         1  28/08/23  11/03/2024  11/03/24  35/2023         2   

         UF_NOTIF  MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  \
0              21    210290      9505  498068.0  ...                  NaN   
1              21    210232     10033  232514.0  ...                  NaN   
2              21    210232     10032       NaN  ...                  NaN   
3              21    210232     10032  232516.0  ...                  NaN   
4              21    210232    100321       NaN  ...                  NaN   
...           ...       ...       ...       ...  ...                  ...   
6816637        13    130020        49     618.0  ...                  NaN   
6816638        13    130406        46      64.0  ...                  NaN   
6816639        13    130406        24     339.0  ...                  NaN   
6816640        16    160025         1  352826.0  ...                  NaN   
6816641        17    172120   1721201    1789.0  ...                  NaN   

         EXAMINADOR ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  \
0               NaN       1         5          NaN           NaN   
1               NaN       2         6          NaN           NaN   
2               NaN       2       NaN          NaN           NaN   
3               NaN       1         6          NaN           NaN   
4               NaN       2         4          NaN           NaN   
...             ...     ...       ...          ...           ...   
6816637       618.0       2         6          NaN           NaN   
6816638        67.0       2         6          NaN           NaN   
6816639       449.0       2         6          NaN           NaN   
6816640    352826.0       2         6          NaN           NaN   
6816641      1789.0       2         6          NaN           NaN   

         DIMEA_AMAMENT  VIVAX  FALCIPARUM  NIV_ESCO  
0                  NaN    NaN         NaN       0.0  
1                  NaN    NaN         NaN       1.0  
2                  NaN    NaN         NaN       0.0  
3                  NaN    NaN         NaN       NaN  
4                  NaN    NaN         NaN       0.0  
...                ...    ...         ...       ...  
6816637            NaN    2.0         2.0       1.0  
6816638            NaN    2.0         2.0       0.0  
6816639            NaN    2.0         2.0       2.0  
6816640            NaN    2.0         2.0       5.0  
6816641            NaN    2.0         2.0       6.0  

[5508409 rows x 48 columns]

In [6]:
len(vivax_cases_df)/len(full_malaria_df)*100

80.80824840148566

About 81% of all malaria cases between 2023 and 2023 were caused by P. Vivax

#### Gets name of UFs

In [7]:
uf_dict = {11 : 'RO',
           12 : 'AC',
           13 : 'AM',
           14 : 'RR',
           15 : 'PA',
           16 : 'AP',
           17 : 'TO',
           21 : 'MA',
           22 : 'PI',
           23 : 'CE',
           24 : 'RN',
           25 : 'PB',
           26 : 'PE',
           27 : 'AL',
           28 : 'SE',
           29 : 'BA',
           31 : 'MG', 
           32 : 'ES',
           33 : 'RJ',
           35 : 'SP',
           41 : 'PR',
           42 : 'SC',
           43 : 'RS',
           50 : 'MS',
           51 : 'MT',
           52 : 'GO',
           53 : 'DF'
          }
corrected_ufs_df = vivax_cases_df.replace({"UF_NOTIF": uf_dict, "UF_RESID": uf_dict})
corrected_ufs_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
0              56  27/01/03  23/03/2011       NaN  05/2003         3       MA   
1             470  20/03/03  23/03/2011       NaN  12/2003         2       MA   
2            1309  10/07/03  23/03/2011       NaN  28/2003         1       MA   
3              19  27/01/03  23/03/2011       NaN  05/2003         3       MA   
4             657  04/04/03  23/03/2011       NaN  14/2003         1       MA   
...           ...       ...         ...       ...      ...       ...      ...   
6816637       225  20/07/23  26/09/2023  26/09/23  29/2023         1       AM   
6816638       524  19/06/23  11/07/2023  11/07/23  25/2023         1       AM   
6816639       922  05/12/23  12/12/2023  12/12/23  49/2023         1       AM   
6816640        14  26/09/23  20/10/2023  20/10/23  39/2023         1       AP   
6816641         1  28/08/23  11/03/2024  11/03/24  35/2023         2       TO   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
0          210290      9505  498068.0  ...                  NaN         NaN   
1          210232     10033  232514.0  ...                  NaN         NaN   
2          210232     10032       NaN  ...                  NaN         NaN   
3          210232     10032  232516.0  ...                  NaN         NaN   
4          210232    100321       NaN  ...                  NaN         NaN   
...           ...       ...       ...  ...                  ...         ...   
6816637    130020        49     618.0  ...                  NaN       618.0   
6816638    130406        46      64.0  ...                  NaN        67.0   
6816639    130406        24     339.0  ...                  NaN       449.0   
6816640    160025         1  352826.0  ...                  NaN    352826.0   
6816641    172120   1721201    1789.0  ...                  NaN      1789.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
0             1         5          NaN           NaN            NaN   NaN   
1             2         6          NaN           NaN            NaN   NaN   
2             2       NaN          NaN           NaN            NaN   NaN   
3             1         6          NaN           NaN            NaN   NaN   
4             2         4          NaN           NaN            NaN   NaN   
...         ...       ...          ...           ...            ...   ...   
6816637       2         6          NaN           NaN            NaN   2.0   
6816638       2         6          NaN           NaN            NaN   2.0   
6816639       2         6          NaN           NaN            NaN   2.0   
6816640       2         6          NaN           NaN            NaN   2.0   
6816641       2         6          NaN           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
0               NaN       0.0  
1               NaN       1.0  
2               NaN       0.0  
3               NaN       NaN  
4               NaN       0.0  
...             ...       ...  
6816637         2.0       1.0  
6816638         2.0       0.0  
6816639         2.0       2.0  
6816640         2.0       5.0  
6816641         2.0       6.0  

[5508409 rows x 48 columns]

#### Gets data for the Amazon region

In [8]:
amazon_df = corrected_ufs_df[corrected_ufs_df.get('UF_NOTIF').isin(['RO', 'AC', 'AM', 'RR', 'PA', 'AP', 'TO'])]
amazon_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
13             72  13/10/03  23/03/2011  14/10/03  42/2003         2       AM   
16             43  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
17             46  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
18             73  03/11/03  23/03/2011  04/11/03  45/2003         2       AM   
19            128  01/04/03  23/03/2011       NaN  14/2003         2       AM   
...           ...       ...         ...       ...      ...       ...      ...   
6816637       225  20/07/23  26/09/2023  26/09/23  29/2023         1       AM   
6816638       524  19/06/23  11/07/2023  11/07/23  25/2023         1       AM   
6816639       922  05/12/23  12/12/2023  12/12/23  49/2023         1       AM   
6816640        14  26/09/23  20/10/2023  20/10/23  39/2023         1       AP   
6816641         1  28/08/23  11/03/2024  11/03/24  35/2023         2       TO   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
13         130260        46     653.0  ...                  NaN         NaN   
16         130260        46     647.0  ...                  NaN         NaN   
17         130260        46     647.0  ...                  NaN         NaN   
18         130260        44    1281.0  ...                  NaN         NaN   
19         130260        44       NaN  ...                  NaN         NaN   
...           ...       ...       ...  ...                  ...         ...   
6816637    130020        49     618.0  ...                  NaN       618.0   
6816638    130406        46      64.0  ...                  NaN        67.0   
6816639    130406        24     339.0  ...                  NaN       449.0   
6816640    160025         1  352826.0  ...                  NaN    352826.0   
6816641    172120   1721201    1789.0  ...                  NaN      1789.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
13            2         6          NaN           NaN            NaN   NaN   
16            2         6          NaN           NaN            NaN   NaN   
17            2         6          NaN           NaN            NaN   NaN   
18            2         5          NaN           NaN            NaN   NaN   
19            2       NaN          NaN           NaN            NaN   NaN   
...         ...       ...          ...           ...            ...   ...   
6816637       2         6          NaN           NaN            NaN   2.0   
6816638       2         6          NaN           NaN            NaN   2.0   
6816639       2         6          NaN           NaN            NaN   2.0   
6816640       2         6          NaN           NaN            NaN   2.0   
6816641       2         6          NaN           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
13              NaN       1.0  
16              NaN       1.0  
17              NaN       1.0  
18              NaN       3.0  
19              NaN      10.0  
...             ...       ...  
6816637         2.0       1.0  
6816638         2.0       0.0  
6816639         2.0       2.0  
6816640         2.0       5.0  
6816641         2.0       6.0  

[5357677 rows x 48 columns]

In [9]:
amazonas_df = amazon_df[amazon_df.get('UF_NOTIF')=='AM']
amazonas_df

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
13             72  13/10/03  23/03/2011  14/10/03  42/2003         2       AM   
16             43  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
17             46  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
18             73  03/11/03  23/03/2011  04/11/03  45/2003         2       AM   
19            128  01/04/03  23/03/2011       NaN  14/2003         2       AM   
...           ...       ...         ...       ...      ...       ...      ...   
6816622      2257  25/09/23  26/09/2023  26/09/23  39/2023         1       AM   
6816624      2261  07/08/23  10/08/2023  10/08/23  32/2023         1       AM   
6816637       225  20/07/23  26/09/2023  26/09/23  29/2023         1       AM   
6816638       524  19/06/23  11/07/2023  11/07/23  25/2023         1       AM   
6816639       922  05/12/23  12/12/2023  12/12/23  49/2023         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
13         130260        46     653.0  ...                  NaN         NaN   
16         130260        46     647.0  ...                  NaN         NaN   
17         130260        46     647.0  ...                  NaN         NaN   
18         130260        44    1281.0  ...                  NaN         NaN   
19         130260        44       NaN  ...                  NaN         NaN   
...           ...       ...       ...  ...                  ...         ...   
6816622    130260       414     244.0  ...                 15,4      3160.0   
6816624    130020        20      29.0  ...                  NaN       460.0   
6816637    130020        49     618.0  ...                  NaN       618.0   
6816638    130406        46      64.0  ...                  NaN        67.0   
6816639    130406        24     339.0  ...                  NaN       449.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
13            2         6          NaN           NaN            NaN   NaN   
16            2         6          NaN           NaN            NaN   NaN   
17            2         6          NaN           NaN            NaN   NaN   
18            2         5          NaN           NaN            NaN   NaN   
19            2       NaN          NaN           NaN            NaN   NaN   
...         ...       ...          ...           ...            ...   ...   
6816622       2         6          NaN           NaN            NaN   2.0   
6816624       2         5          2.0           NaN            NaN   2.0   
6816637       2         6          NaN           NaN            NaN   2.0   
6816638       2         6          NaN           NaN            NaN   2.0   
6816639       2         6          NaN           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
13              NaN       1.0  
16              NaN       1.0  
17              NaN       1.0  
18              NaN       3.0  
19              NaN      10.0  
...             ...       ...  
6816622         2.0       6.0  
6816624         2.0       6.0  
6816637         2.0       1.0  
6816638         2.0       0.0  
6816639         2.0       2.0  

[2013685 rows x 48 columns]

In [10]:
len(amazonas_df)/len(vivax_cases_df)*100

36.55656288412861

About 36.5% of all malaria cases caused by P. Vivax between 2023 and 2023 were notified in the state of Amazonas

In [11]:
amazonas_df['UF_NOTIF'].unique()

array(['AM'], dtype=object)

In [12]:
amazonas_df['UF_RESID'].unique()

array(['AM', 'RO', 'MT', 'AC', 'PA', nan, 'RR', 'MA', 'RS', 'AP', 'ES',
       'DF', 'PI', 'MG', 'BA', 'TO', 'GO', 'PE', 'AL', 'PR', 'MS', 'SC',
       'SP', 'RJ', 'RN'], dtype=object)

In [13]:
amazonas_notif_and_resid = amazon_df[(amazon_df.get('UF_NOTIF')=='AM') & (amazon_df.get('UF_RESID')=='AM')]
amazonas_notif_and_resid

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
13             72  13/10/03  23/03/2011  14/10/03  42/2003         2       AM   
16             43  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
17             46  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
18             73  03/11/03  23/03/2011  04/11/03  45/2003         2       AM   
19            128  01/04/03  23/03/2011       NaN  14/2003         2       AM   
...           ...       ...         ...       ...      ...       ...      ...   
6816621      1019  30/10/23  01/11/2023  01/11/23  44/2023         1       AM   
6816622      2257  25/09/23  26/09/2023  26/09/23  39/2023         1       AM   
6816624      2261  07/08/23  10/08/2023  10/08/23  32/2023         1       AM   
6816638       524  19/06/23  11/07/2023  11/07/23  25/2023         1       AM   
6816639       922  05/12/23  12/12/2023  12/12/23  49/2023         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
13         130260        46     653.0  ...                  NaN         NaN   
16         130260        46     647.0  ...                  NaN         NaN   
17         130260        46     647.0  ...                  NaN         NaN   
18         130260        44    1281.0  ...                  NaN         NaN   
19         130260        44       NaN  ...                  NaN         NaN   
...           ...       ...       ...  ...                  ...         ...   
6816621    130060         4      57.0  ...                  NaN        11.0   
6816622    130260       414     244.0  ...                 15,4      3160.0   
6816624    130020        20      29.0  ...                  NaN       460.0   
6816638    130406        46      64.0  ...                  NaN        67.0   
6816639    130406        24     339.0  ...                  NaN       449.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
13            2         6          NaN           NaN            NaN   NaN   
16            2         6          NaN           NaN            NaN   NaN   
17            2         6          NaN           NaN            NaN   NaN   
18            2         5          NaN           NaN            NaN   NaN   
19            2       NaN          NaN           NaN            NaN   NaN   
...         ...       ...          ...           ...            ...   ...   
6816621       1         6          NaN           NaN            NaN   1.0   
6816622       2         6          NaN           NaN            NaN   2.0   
6816624       2         5          2.0           NaN            NaN   2.0   
6816638       2         6          NaN           NaN            NaN   2.0   
6816639       2         6          NaN           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
13              NaN       1.0  
16              NaN       1.0  
17              NaN       1.0  
18              NaN       3.0  
19              NaN      10.0  
...             ...       ...  
6816621         2.0       5.0  
6816622         2.0       6.0  
6816624         2.0       6.0  
6816638         2.0       0.0  
6816639         2.0       2.0  

[2004500 rows x 48 columns]

In [14]:
len(amazonas_notif_and_resid)/len(amazonas_df)*100

99.5438710622565

From all cases notified in the state of Amazonas, about 99.5% of the infected were residents of the state.

In [18]:
cases_by_AM_municipality = amazonas_notif_and_resid[['MUN_RESI']].groupby(['MUN_RESI']).size().reset_index(name='TOTAL_CASES').sort_values(by='TOTAL_CASES', ascending=False)
cases_by_AM_municipality['MUN_RESI'] = cases_by_AM_municipality['MUN_RESI'].astype(int)
cases_by_AM_municipality

MUN_RESI  TOTAL_CASES
37    130260       521362
51    130380       134459
7     130040        92387
19    130120        77098
57    130420        72717
..       ...          ...
58    130423          896
3     130010          505
8     130050          182
11    130068          177
61    130440          155

[62 rows x 2 columns]

It is possible to see that Manaus is the city of residence of most notified cases in the state of Amazonas

In [20]:
manaus_notif_and_resid = amazonas_notif_and_resid[(amazonas_notif_and_resid.get('MUN_NOTI')==130260) & (amazonas_notif_and_resid.get('MUN_RESI')==130260)]
manaus_notif_and_resid

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
13             72  13/10/03  23/03/2011  14/10/03  42/2003         2       AM   
16             43  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
17             46  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
18             73  03/11/03  23/03/2011  04/11/03  45/2003         2       AM   
19            128  01/04/03  23/03/2011       NaN  14/2003         2       AM   
...           ...       ...         ...       ...      ...       ...      ...   
6816487      8400  20/12/23  21/12/2023  21/12/23  51/2023         2       AM   
6816566        95  26/09/23  09/10/2023  09/10/23  39/2023         1       AM   
6816585       130  24/12/23  28/12/2023  28/12/23  52/2023         1       AM   
6816608       125  17/12/23  21/12/2023  21/12/23  51/2023         1       AM   
6816622      2257  25/09/23  26/09/2023  26/09/23  39/2023         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
13         130260        46     653.0  ...                  NaN         NaN   
16         130260        46     647.0  ...                  NaN         NaN   
17         130260        46     647.0  ...                  NaN         NaN   
18         130260        44    1281.0  ...                  NaN         NaN   
19         130260        44       NaN  ...                  NaN         NaN   
...           ...       ...       ...  ...                  ...         ...   
6816487    130260        19    2745.0  ...                 17,5        50.0   
6816566    130260       103    4317.0  ...                  NaN      4317.0   
6816585    130260       103    4468.0  ...                  NaN      4468.0   
6816608    130260       103    4913.0  ...                  NaN      4913.0   
6816622    130260       414     244.0  ...                 15,4      3160.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
13            2         6          NaN           NaN            NaN   NaN   
16            2         6          NaN           NaN            NaN   NaN   
17            2         6          NaN           NaN            NaN   NaN   
18            2         5          NaN           NaN            NaN   NaN   
19            2       NaN          NaN           NaN            NaN   NaN   
...         ...       ...          ...           ...            ...   ...   
6816487       2         6          NaN           NaN            NaN   2.0   
6816566       2         6          NaN           NaN            NaN   2.0   
6816585       2         6          NaN           NaN            NaN   2.0   
6816608       2         6          NaN           NaN            NaN   2.0   
6816622       2         6          NaN           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
13              NaN       1.0  
16              NaN       1.0  
17              NaN       1.0  
18              NaN       3.0  
19              NaN      10.0  
...             ...       ...  
6816487         2.0       5.0  
6816566         2.0      10.0  
6816585         2.0       1.0  
6816608         2.0       1.0  
6816622         2.0       6.0  

[516683 rows x 48 columns]

In [21]:
manaus_notif = amazonas_notif_and_resid[amazonas_notif_and_resid.get('MUN_NOTI')==130260]
manaus_notif

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
13             72  13/10/03  23/03/2011  14/10/03  42/2003         2       AM   
16             43  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
17             46  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
18             73  03/11/03  23/03/2011  04/11/03  45/2003         2       AM   
19            128  01/04/03  23/03/2011       NaN  14/2003         2       AM   
...           ...       ...         ...       ...      ...       ...      ...   
6816487      8400  20/12/23  21/12/2023  21/12/23  51/2023         2       AM   
6816566        95  26/09/23  09/10/2023  09/10/23  39/2023         1       AM   
6816585       130  24/12/23  28/12/2023  28/12/23  52/2023         1       AM   
6816608       125  17/12/23  21/12/2023  21/12/23  51/2023         1       AM   
6816622      2257  25/09/23  26/09/2023  26/09/23  39/2023         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
13         130260        46     653.0  ...                  NaN         NaN   
16         130260        46     647.0  ...                  NaN         NaN   
17         130260        46     647.0  ...                  NaN         NaN   
18         130260        44    1281.0  ...                  NaN         NaN   
19         130260        44       NaN  ...                  NaN         NaN   
...           ...       ...       ...  ...                  ...         ...   
6816487    130260        19    2745.0  ...                 17,5        50.0   
6816566    130260       103    4317.0  ...                  NaN      4317.0   
6816585    130260       103    4468.0  ...                  NaN      4468.0   
6816608    130260       103    4913.0  ...                  NaN      4913.0   
6816622    130260       414     244.0  ...                 15,4      3160.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
13            2         6          NaN           NaN            NaN   NaN   
16            2         6          NaN           NaN            NaN   NaN   
17            2         6          NaN           NaN            NaN   NaN   
18            2         5          NaN           NaN            NaN   NaN   
19            2       NaN          NaN           NaN            NaN   NaN   
...         ...       ...          ...           ...            ...   ...   
6816487       2         6          NaN           NaN            NaN   2.0   
6816566       2         6          NaN           NaN            NaN   2.0   
6816585       2         6          NaN           NaN            NaN   2.0   
6816608       2         6          NaN           NaN            NaN   2.0   
6816622       2         6          NaN           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
13              NaN       1.0  
16              NaN       1.0  
17              NaN       1.0  
18              NaN       3.0  
19              NaN      10.0  
...             ...       ...  
6816487         2.0       5.0  
6816566         2.0      10.0  
6816585         2.0       1.0  
6816608         2.0       1.0  
6816622         2.0       6.0  

[522947 rows x 48 columns]

In [22]:
manaus_resid = amazonas_notif_and_resid[amazonas_notif_and_resid.get('MUN_RESI')==130260]
manaus_resid

COD_NOTI  DT_NOTIF    DT_ENVLO  DT_DIGIT SEM_NOTI  TIPO_LAM UF_NOTIF  \
13             72  13/10/03  23/03/2011  14/10/03  42/2003         2       AM   
16             43  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
17             46  10/10/03  23/03/2011  14/10/03  41/2003         2       AM   
18             73  03/11/03  23/03/2011  04/11/03  45/2003         2       AM   
19            128  01/04/03  23/03/2011       NaN  14/2003         2       AM   
...           ...       ...         ...       ...      ...       ...      ...   
6816487      8400  20/12/23  21/12/2023  21/12/23  51/2023         2       AM   
6816566        95  26/09/23  09/10/2023  09/10/23  39/2023         1       AM   
6816585       130  24/12/23  28/12/2023  28/12/23  52/2023         1       AM   
6816608       125  17/12/23  21/12/2023  21/12/23  51/2023         1       AM   
6816622      2257  25/09/23  26/09/2023  26/09/23  39/2023         1       AM   

         MUN_NOTI  COD_UNIN  COD_AGEN  ... NU_HEMOGLOBINA_TOTAL  EXAMINADOR  \
13         130260        46     653.0  ...                  NaN         NaN   
16         130260        46     647.0  ...                  NaN         NaN   
17         130260        46     647.0  ...                  NaN         NaN   
18         130260        44    1281.0  ...                  NaN         NaN   
19         130260        44       NaN  ...                  NaN         NaN   
...           ...       ...       ...  ...                  ...         ...   
6816487    130260        19    2745.0  ...                 17,5        50.0   
6816566    130260       103    4317.0  ...                  NaN      4317.0   
6816585    130260       103    4468.0  ...                  NaN      4468.0   
6816608    130260       103    4913.0  ...                  NaN      4913.0   
6816622    130260       414     244.0  ...                 15,4      3160.0   

        ID_LVC  GESTANTE   AMAMENTANDO  TEMP_AMAMENT  DIMEA_AMAMENT VIVAX  \
13            2         6          NaN           NaN            NaN   NaN   
16            2         6          NaN           NaN            NaN   NaN   
17            2         6          NaN           NaN            NaN   NaN   
18            2         5          NaN           NaN            NaN   NaN   
19            2       NaN          NaN           NaN            NaN   NaN   
...         ...       ...          ...           ...            ...   ...   
6816487       2         6          NaN           NaN            NaN   2.0   
6816566       2         6          NaN           NaN            NaN   2.0   
6816585       2         6          NaN           NaN            NaN   2.0   
6816608       2         6          NaN           NaN            NaN   2.0   
6816622       2         6          NaN           NaN            NaN   2.0   

         FALCIPARUM  NIV_ESCO  
13              NaN       1.0  
16              NaN       1.0  
17              NaN       1.0  
18              NaN       3.0  
19              NaN      10.0  
...             ...       ...  
6816487         2.0       5.0  
6816566         2.0      10.0  
6816585         2.0       1.0  
6816608         2.0       1.0  
6816622         2.0       6.0  

[521362 rows x 48 columns]